youtube_dl/extractor/youporn.py

```diff
@@ -66,57 +73,108 @@ class YouPornIE(InfoExtractor):
     }, {
         'url': 'https://www.youporn.com/watch/13922959/femdom-principal/',
         'only_matching': True,
+    }, {
+        'url': 'https://www.youporn.com/watch/16290308/tinderspecial-trailer1/',
+        'info_dict': {
+            'id': '46949121',
+            'age_limit': 18,
+            'categories': [],
+            'description': None,  # SEO spam using title removed
+            'display_id': 'tinderspecial-trailer1',
+            'duration': 298.0,
+            'ext': 'mp4',
+            'upload_date': '20201123',
+            'uploader': 'Ersties',
+            'tags': [],
+            'thumbnail': 'https://fi1.ypncdn.com/m=eaSaaTbWx/202011/23/16290308/original/3.jpg',
+            'timestamp': 1606147564,
+            'title': 'Tinder In Real Life',
+            'view_count': int,
+        }
     }]
 
-    @staticmethod
-    def _extract_urls(webpage):
-        return re.findall(
-            r'<iframe[^>]+\bsrc=["\']((?:https?:)?//(?:www\.)?youporn\.com/embed/\d+)',
-            webpage)
+    @classmethod
+    def _extract_urls(cls, webpage):
+        def yield_urls():
+            for p in cls._EMBED_REGEX:
+                for m in re.finditer(p, webpage):
+                    yield m.group('url')
 
-    def _real_extract(self, url):
-        mobj = re.match(self._VALID_URL, url)
-        video_id = mobj.group('id')
-        display_id = mobj.group('display_id') or video_id
+        return list(yield_urls())
 
-        definitions = self._download_json(
-            'https://www.youporn.com/api/video/media_definitions/%s/' % video_id,
-            display_id)
+    def _real_extract(self, url):
+        # Actual video ID is hidden in the page
+        display_id = self._match_valid_url(url).group('id', 'display_id')
+        url = 'http://www.youporn.com/watch/%s' % (display_id[0],)
+        display_id = display_id[1] or display_id[0]
+        webpage = self._download_webpage(
+            url, display_id, headers={'Cookie': 'age_verified=1'})
+
+        watchable = self._search_regex(
+            r'''(<div\s[^>]*\bid\s*=\s*('|")?watch-container(?(2)\2|(?!-)\b)[^>]*>)''',
+            webpage, 'video id', default=None)
+        if not watchable:
+            msg = re.split(r'\s{4}', clean_html(get_element_by_id(
+                'mainContent', webpage)) or '')[0]
+            raise ExtractorError(
+                ('%s says: %s' % (self.IE_NAME, msg))
+                if msg else 'Video unavailable: no reason found',
+                expected=True)
+        video_id = extract_attributes(watchable).get('data-video-id')
+        if not video_id:
+            # last chance saloon
+            video_id = self._search_regex(
+                r'''\bcurrentVideoId\s*:\s*('|")(?P<id>\d+)\1''',
+                webpage, 'video id', group='id')
```

Comment:
This gets an/the internal video ID that is not revealed in the URL or displayed page content. I used this since it appears to index the actual media items on the site (ie, there could be two pages with different URL IDs but identical media content), but still have doubts.

Reply:
As I couldn't see this ID being used in the site code, I reverted to the ID from the URL.

Reply:
Has this fix been confirmed to actually work? I still get a JSON metadata 404 with v2024.04.28.232723.

Reply:
> please open a broken site issue 

Quick question before I do open a new issue: 

How am I supposed to make sure this code is indeed included in the yt-dlp version that I download in [Seal](https://github.com/JunkFood02/Seal) or [YTDLnis](https://github.com/deniscerri/ytdlnis) and run, other than verifying it's the lates master (2024.04.28.221944) or nightly (2024.04.28.232723) version? 

Just below the title of this PR, there's this info 
> [dirkf](https://github.com/dirkf) merged 4 commits into [ytdl-org:master](https://github.com/ytdl-org/youtube-dl/tree/master) from [dirkf:df-youporn-extractor-ovrhaul](https://github.com/dirkf/youtube-dl/tree/df-youporn-extractor-ovrhaul) last week

Despite the [annoying vagueness](https://github.com/orgs/community/discussions/5972#discussion-3607847) of "last week", it seems to be clear in this case that both versions date later than the merge, or am I mistaken?